### PR TITLE
gh-98713: Use `@cpython_only` for a test that fails on PyPy

### PR DESCRIPTION
**Pull request by sobolevn**
* Issue: gh-98713

Automerge-Triggered-By: GH:AlexWaygood

### CODE REVIEW

**Review by AlexWaygood — COMMENTED** [review]
Should we maybe add a short news entry? The test made it into 3.11.0.

Otherwise this looks good! You beat the PR I was writing to the finish line ;)

**Comment by sobolevn** [comment]
I've never seen a NEWS entry for tests. If others say so - I surely will add it.

**Comment by AlexWaygood** [comment]
We usually skip news for PRs that improve test coverage etc, but this is fixing a bug in the test suite that existed in a previous release of Python, so I'd argue for including a short news entry.

File: Misc/NEWS.d/next/Tests/2022-10-26-15-19-20.gh-issue-98713.Lnu32R.rst [hunk 1]
```diff
@@ -0,0 +1,3 @@
+Fix a bug in test suite when cpython-specific implementation details were
+not covered by ``@cpython_only`` and were not skipped on other
+implementations.
```
Comment (AlexWaygood) [inline Misc/NEWS.d/next/Tests/2022-10-26-15-19-20.gh-issue-98713.Lnu32R.rst]:
```suggestion
Fix a bug in the :mod:`typing` tests where a test relying on CPython-specific
implementation details was not decorated with ``@cpython_only`` and was not
skipped on other implementations.
```

**Review by AlexWaygood — APPROVED** [review]

**Review by kumaraditya303 — APPROVED** [review]

**Comment by AlexWaygood** [comment]
The `test_asyncio` failure is unrelated and will be fixed by #98704.

**Pull request merged**